sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,232 @@
+from sympy import sin, cos, Matrix, sqrt, pi, expand_mul, Abs, acos, S
+from sympy.core.symbol import symbols
+from sympy.physics.mechanics import dynamicsymbols, Body, PinJoint
+from sympy.physics.mechanics.joint import Joint
+from sympy.physics.vector import Vector, ReferenceFrame
+from sympy.testing.pytest import raises
+
+t = dynamicsymbols._t
+
+
+def test_Joint():
+    parent = Body('parent')
+    child = Body('child')
+    raises(TypeError, lambda: Joint('J', parent, child))
+
+
+def test_pinjoint():
+    P = Body('P')
+    C = Body('C')
+    l, m = symbols('l m')
+    theta, omega = dynamicsymbols('theta_J, omega_J', real=True)
+
+    Pj = PinJoint('J', P, C )
+    assert Pj._name == 'J'
+    assert Pj.parent() == P
+    assert Pj.child() == C
+    assert Pj.coordinates() == [theta]
+    assert Pj.speeds() == [omega]
+    assert Pj.kdes() == [omega - theta.diff(t)]
```

Comment:
For future multi degree of freedom joints, I think this should provide the solved equations. If this gave `{theta.diff(t): omega}` it would